### DISCUSSION TITLE
Is a purely "offline" conversion of a `Jenkinsfile` possible?

### DISCUSSION THREAD

**Discussion by sschuberth** [github/gh-actions-importer#57] [Feedback] [answered]
I tried running something like

    gh actions-importer dry-run jenkins --source-file-path integrations/jenkins/Jenkinsfile --output-dir gh-action

but got

    Option '--source-url' is required.

Just wondering, why is access to "The URL of the Jenkins job to migrate" actually required? Shouldn't all required information for a conversion present in `Jenkinsfile` alone?

**Comment by ethanis** [comment, marked as answer]
@sschuberth there is not a way to have a purely offline conversion. The Jenkins server is required to be used, even when the `--source-file-path` option is being used, in order to convert the Jenkinsfile into an abstract syntax tree that can be converted by Actions Importer. The endpoint that is used is the [pipeline-model-converter/toJson](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/EXTENDING.md#conversion-to-json-representation-from-jenkinsfile) endpoint.
